### ISSUE TITLE
AssertionError in GSUB (or GPOS) table

### ISSUE THREAD

**Issue by silicontrip** [behdad/fonttools#344]
I receive the following error when trying to compile a TTX file into a TTF for some fonts. 
If I run without -v It says the GSUB table.

`Parsing 'GSUB' table...`
`Traceback (most recent call last):`

If I run with -v it says the GPOS table.
`compiling 'GPOS' table  (15:36:24)`
`Traceback (most recent call last):`

I'm not sure how to attach the xml of those tables.
Thanks for your help

Full stack trace here

`Traceback (most recent call last):`
`  File "/usr/local/bin/ttx", line 9, in <module>`
`    load_entry_point('fonttools==2.5', 'console_scripts', 'ttx')()`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttx.py", line 350, in main`
`    process(jobs, options)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttx.py", line 333, in process`
`    action(input, output, options)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttx.py", line 249, in ttCompile`
`    ttf.save(output)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/__init__.py", line 221, in save`
`    self._writeTable(tag, writer, done)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/__init__.py", line 653, in _writeTable`
`    tabledata = self.getTableData(tag)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/__init__.py", line 666, in getTableData`
`    return self.tables[tag].compile(self)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otBase.py", line 85, in compile`
`    self.table.compile(writer, font)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otBase.py", line 657, in compile`
`    conv.write(writer, font, table, value)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otConverters.py", line 349, in write`
`    value.compile(subWriter, font)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otBase.py", line 637, in compile`
`    conv.writeArray(writer, font, table, value)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otConverters.py", line 124, in writeArray`
`    self.write(writer, font, tableDict, values[i], i)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otConverters.py", line 349, in write`
`    value.compile(subWriter, font)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otBase.py", line 637, in compile`
`    conv.writeArray(writer, font, table, value)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otConverters.py", line 124, in writeArray`
`    self.write(writer, font, tableDict, values[i], i)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otConverters.py", line 349, in write`
`    value.compile(subWriter, font)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otBase.py", line 657, in compile`
`    conv.write(writer, font, table, value)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otConverters.py", line 349, in write`
`    value.compile(subWriter, font)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otBase.py", line 637, in compile`
`    conv.writeArray(writer, font, table, value)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otConverters.py", line 124, in writeArray`
`    self.write(writer, font, tableDict, values[i], i)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otConverters.py", line 273, in write`
`    value.compile(writer, font)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otBase.py", line 636, in compile`
`    writer[conv.repeat].setValue(countValue)`
`  File "/Library/Python/2.7/site-packages/fonttools-2.5-py2.7.egg/fontTools/ttLib/tables/otBase.py", line 524, in setValue`
`    assert table[name] == value, (name, table[name], value)`
`AssertionError: ('ClassCount', 2, 1)`

**Comment by anthrotype** [comment]
@silicontrip are these fonts available somewhere, so that we can try to replicate the issue?

**Comment by silicontrip** [comment]
Sorry about that.  I was in the process of cleaning out corrupt fonts and I assume this one got deleted with it. I've tried going back to the source of fonts but cannot find any that reproduce the problem.   It is possible that the ttx file was produced with an earlier version of fonttools.   (i've been tripped up a few times with the auto incrementing file name writer)

I have found a few (corrupt) fonts that fail to produce valid XML, by causing an early exit.  Are you interested in those?
